contracts/src/main/kotlin/jp/datachain/corda/ibc/ics25/Handler.kt

```diff
@@ -23,41 +23,34 @@ import jp.datachain.corda.ibc.ics23.CommitmentProof
 import jp.datachain.corda.ibc.ics24.Identifier
 import jp.datachain.corda.ibc.ics26.Context
 import jp.datachain.corda.ibc.states.IbcChannel
+import jp.datachain.corda.ibc.states.IbcClientState
 import jp.datachain.corda.ibc.states.IbcConnection
 import jp.datachain.corda.ibc.types.Timestamp
 
 object Handler {
     fun createClient(ctx: Context, msg: ClientTx.MsgCreateClient) {
         val host = ctx.getInput<Host>()
-
-        when {
-            msg.clientState.`is`(Corda.ClientState::class.java) -> {
-                val (nextHost, clientId) = host.generateClientIdentifier(ClientType.CordaClient)
-                val clientState = msg.clientState.unpack(Corda.ClientState::class.java)!!
-                val consensusState = msg.consensusState.unpack(Corda.ConsensusState::class.java)!!
-                ctx.addOutput(CordaClientState(host, clientId, clientState, consensusState))
-                ctx.addOutput(nextHost)
-            }
-            msg.clientState.`is`(Fabric.ClientState::class.java) -> {
-                val (nextHost, clientId) = host.generateClientIdentifier(ClientType.FabricClient)
-                val clientState = msg.clientState.unpack(Fabric.ClientState::class.java)!!
-                val consensusState = msg.consensusState.unpack(Fabric.ConsensusState::class.java)!!
-                ctx.addOutput(FabricClientState(host, clientId, clientState, consensusState))
-                ctx.addOutput(nextHost)
-            }
+        val clientState: ClientState = when {
+            msg.clientState.`is`(Corda.ClientState::class.java) -> CordaClientState(msg.clientState, msg.consensusState)
+            msg.clientState.`is`(Fabric.ClientState::class.java) -> FabricClientState(msg.clientState, msg.consensusState)
             msg.clientState.`is`(Tendermint.ClientState::class.java) -> throw NotImplementedError()
             msg.clientState.`is`(Solomachine.ClientState::class.java) -> throw NotImplementedError()
             msg.clientState.`is`(Localhost.ClientState::class.java) -> throw NotImplementedError()
             else -> throw IllegalArgumentException()
         }
+
+        val (nextHost, clientId) = host.generateClientIdentifier(ClientType.CordaClient)
```

Comment:
looks like client type is always corda. Is it correct?

Reply:
Ouch. This is simply a bug. I fixed it. Please wait for CI done.